### PR TITLE
Adopt new workflow in campaign detail view

### PR DESCRIPTION
**Pull request by eseliger**
Applies the first round of new designs from the campaigns figma workbook. Also restructures the components to better fit their new use-case.

### CODE REVIEW

File: web/src/enterprise/campaigns/apply/ChangesetSpecList.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .changeset-spec-list {
     &__grid {
         display: grid;
-        grid-template-columns: [caret] min-content [action] min-content [info] auto [diffstat] min-content [end];
+        grid-template-columns: [caret] min-content [action] min-content [info] minmax(auto, 1fr) [diffstat] min-content [end];
```
Comment (eseliger) [inline web/src/enterprise/campaigns/apply/ChangesetSpecList.scss:4]: To make it even more mind-blowing with the grid usage @LawnGnome:
The fix for overflowing large diffs

File: web/src/enterprise/campaigns/detail/CampaignStatsCard.tsx [hunk 2]
```diff
@@ -29,23 +40,36 @@ export const CampaignStatsCard: React.FunctionComponent<CampaignStatsCardProps>
...
         </div>
     )
 }
+
+export const CampaignStatsTotalAction: React.FunctionComponent<{ count: number }> = ({ count }) => (
+    <div className="m-0 mr-3 flex-grow-0 flex-shrink-0 text-nowrap d-flex flex-column align-items-center justify-content-center">
+        <span className="campaign-stats-card__changesets-pill">
+            <span className="badge badge-pill badge-secondary">{count}</span>
+        </span>
+        <span className="text-muted">changesets</span>
```
Comment (LawnGnome) [inline web/src/enterprise/campaigns/detail/CampaignStatsCard.tsx:73]: I think this is another pluraliser case. (Sorry.)

File: web/src/enterprise/campaigns/detail/changesets/HiddenExternalChangesetNode.tsx [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
+                </h3>
+            </div>
+            <div>
+                <ChangesetLastSynced changeset={node} viewerCanAdminister={false} />
             </div>
         </div>
-    </li>
+        <span />
+        <span />
+        <span />
```
Comment (LawnGnome) [inline web/src/enterprise/campaigns/detail/changesets/HiddenExternalChangesetNode.tsx:29]: Do I even want to know about these empty spans (and the one above)?

(Well, OK, I kinda do: these are the placeholders for the state fields that are hidden, right?)

File: web/src/enterprise/campaigns/global/GlobalCampaignsArea.tsx [hunk 4]
```diff
@@ -55,31 +51,13 @@ export const AuthenticatedCampaignsArea = withAuthenticatedUser<AuthenticatedPro
...
-                        render={props => <CampaignCliHelp {...outerProps} {...props} />}
-                        exact={true}
-                    />
-                    <Route
-                        path={`${match.url}/apply/:specID`}
-                        render={({ match, ...props }: RouteComponentProps<{ specID: string }>) => (
-                            <CampaignApplyPage {...outerProps} {...props} specID={match.params.specID} />
-                        )}
-                        exact={true}
-                    />
```
Comment (LawnGnome) [inline web/src/enterprise/campaigns/global/GlobalCampaignsArea.tsx:82]: This is just getting rid of the non-namespaced apply route, right?

Comment (eseliger) [reply web/src/enterprise/campaigns/global/GlobalCampaignsArea.tsx:82]: Correct :)

File: web/src/enterprise/campaigns/detail/changesets/HiddenExternalChangesetNode.tsx (hunk 3 above)
Comment (eseliger) [reply web/src/enterprise/campaigns/detail/changesets/HiddenExternalChangesetNode.tsx:29]: It's for the grid, see them like `td`s in tables :) and yes, it's just empty cells here

File: web/src/enterprise/campaigns/detail/changesets/CampaignChangesetsHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+                Displaying {nodes.length}
+                {totalCount && <> of {totalCount}</>} changesets
+            </strong>
+        </div>
+        <span />
+        <h5 className="text-uppercase text-center text-nowrap text-muted">Status</h5>
+        <h5 className="text-uppercase text-nowrap text-muted">Changeset information</h5>
+        <h5 className="text-uppercase text-center text-nowrap text-muted">Check state</h5>
+        <h5 className="text-uppercase text-center text-nowrap text-muted">Review state</h5>
+        <h5 className="text-uppercase text-right text-nowrap text-muted">Changes</h5>
```
Comment (felixfbecker) [inline web/src/enterprise/campaigns/detail/changesets/CampaignChangesetsHeader.tsx:25]: **issue**: I'm not sure if this is correct for accessibility (whether a screen reader wouldn't expect the content to follow the heading in the DOM tree). There are probably some best practices here, maybe the grid needs `role="table"` and there is some kind of `columnheading` role that could be applied here

File: web/src/enterprise/campaigns/apply/CreateUpdateCampaignAlert.story.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    return (
+        <>
+            <Tooltip />
+            <style>{webStyles}</style>
+            <div className="p-3 container web-content">{story()}</div>
+        </>
+    )
+})
+
+const setIsLoadingStub = () => undefined
```
[the remark this answers is not in the document]
Comment (eseliger) [reply web/src/enterprise/campaigns/apply/CreateUpdateCampaignAlert.story.tsx:22]: Ah true, I didn't think of that 🤦

File: web/src/enterprise/campaigns/detail/CampaignStatsCard.scss [hunk 7]
```diff
@@ -0,0 +1,7 @@
+.campaign-stats-card {
+    &__changesets-pill {
+        // Should be the same height as plain mdi icons.
+        /* stylelint-disable declaration-property-unit-whitelist */
+        height: 24px;
```
[the remark this answers is not in the document]
Comment (eseliger) [reply web/src/enterprise/campaigns/detail/CampaignStatsCard.scss:5]: We don't use icon-inline for the icons that just float without text, as they weren't aligned perfectly
I want to revisit if there's a better way to do this, but needed to get something in for the branch cut :)

**Pull request merged**